core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -245,7 +245,16 @@ private FormValidation verifySignature(JSONObject o) throws IOException {
      */
     @Nonnull
     protected JSONSignatureValidator getJsonSignatureValidator() {
-        return new JSONSignatureValidator("update site '"+id+"'");
+        return getJsonSignatureValidator("update site '"+id+"'");
+    }
+
+    /**
+     * Let sub-classes of UpdateSite provide their own signature validator.
+     * @return the signature validator.
+     */
+    @Nonnull
+    protected JSONSignatureValidator getJsonSignatureValidator(String name) {
```

Comment:
I would deprecate the non-parameterized method and make this one CheckForNull with a fallback to <code>id</code>. Anyway, the inheritance seems to be fragile, not sure why external users need to pass the "update site '" text prefix